source/common/common/logger.cc

```diff
@@ -55,29 +57,43 @@ void DelegatingLogSink::set_formatter(std::unique_ptr<spdlog::formatter> formatt
 
 void DelegatingLogSink::log(const spdlog::details::log_msg& msg) {
   absl::ReleasableMutexLock lock(&format_mutex_);
-  if (!formatter_) {
-    lock.Release();
-    sink_->log(absl::string_view(msg.payload.data(), msg.payload.size()));
-    return;
+  absl::string_view msg_view = absl::string_view(msg.payload.data(), msg.payload.size());
+  if (formatter_) {
+    fmt::memory_buffer formatted;
+    formatter_->format(msg, formatted);
+    msg_view = absl::string_view(formatted.data(), formatted.size());
   }
-
-  fmt::memory_buffer formatted;
-  formatter_->format(msg, formatted);
   lock.Release();
-  sink_->log(absl::string_view(formatted.data(), formatted.size()));
+
+  if (should_escape_) {
+    // Order of the StripSuffix matters: It strips both `\n` and `\r\n` end-of-line characters
+    msg_view = absl::StripSuffix(msg_view, "\n");
+    msg_view = absl::StripSuffix(msg_view, "\r");
+    sink_->log(absl::CEscape(msg_view));
+    sink_->log(spdlog::details::os::default_eol);
```

Comment:
did you see my earlier comment about simply saving the whitespace at EOL and then re-appending the same whitespace?

I think that would be more correct than this one, which might change whether/how-many line-breaks occur.

Reply:
I did this with some substrings on the `absl::string_view`. 

I manually tested this change, but I don't know how to unit test it. We don't have any mocks for the Sinks, so I can't double-check that the escape characters are logged to the underlying sink.

Do you have any suggestions on how to test this code?